data-prepper-plugins/otel-proto-common/src/main/java/org/opensearch/dataprepper/plugins/otel/codec/OTelLogsProtoBufDecoder.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.opensearch.dataprepper.plugins.otel.codec;
+
+import io.opentelemetry.proto.collector.logs.v1.ExportLogsServiceRequest;
+import org.opensearch.dataprepper.model.codec.ByteDecoder;
+import org.opensearch.dataprepper.model.record.Record;
+
+import org.opensearch.dataprepper.model.event.Event;
+import org.opensearch.dataprepper.model.log.OpenTelemetryLog;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+ 
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.function.Consumer;
+import java.time.Instant;
+ 
+public class OTelLogsProtoBufDecoder implements ByteDecoder {
+    private static final Logger LOG = LoggerFactory.getLogger(OTelLogsProtoBufDecoder.class);
+    private static final int MAX_REQUEST_LEN = (8 * 1024 * 1024);
+    private final OTelProtoCodec.OTelProtoDecoder otelProtoDecoder;
+    private final boolean lengthPrefixedEncoding;
+    
+    public OTelLogsProtoBufDecoder(boolean lengthPrefixedEncoding) {
+        otelProtoDecoder = new OTelProtoCodec.OTelProtoDecoder();
+        this.lengthPrefixedEncoding = lengthPrefixedEncoding;
+    }
+
+    private void parseRequest(byte[] buffer, final Instant timeReceivedMs, Consumer<Record<Event>> eventConsumer) throws IOException {
+        ExportLogsServiceRequest request = ExportLogsServiceRequest.parseFrom(buffer);
```

Comment:
Why do you choose to call the method with a byte array? The is a version, that accepts an `InputStream` as well. This might alleviate the need to create large arrays.

Reply:
@KarstenSchnitter as mentioned above, it can only be used in one case. In the other case, we have to read "len" bytes because passing InputStream doesn't indicate how many bytes to read.